plugin-aggregation/src/transform.rs

```diff
@@ -0,0 +1,751 @@
+        let metric_correspondence_table_read = (*metric_correspondence_table_clone)
+            .read()
+            .map_err(|_| anyhow!("could not read the metric correspondence table"))?;
+
+        let mut aggregated_points = MeasurementBuffer::new();
+        log::debug!("buffer size: {}", self.internal_buffer.len());
+
+        for key in self.internal_buffer.clone().keys() {
+            let values = self.internal_buffer.get_mut(&key).ok_or(TransformError::Fatal(anyhow!(
+                "error while trying to read the internal buffer"
+            )))?;
+            // TODO: Clean the internal_buffer by deleting the empty values/key P2.
+            loop {
+                let min_timestamp = compute_min_timestamp(values[0].timestamp, self.interval);
+
+                if !contains_enough_data(self.interval, &values, min_timestamp) {
+                    break;
+                }
+
+                let (i, j) = get_ids(self.interval, &values, min_timestamp)?;
+
+                let sub_vec: Vec<MeasurementPoint> = values.drain(i..=j).collect();
+
+                // Compute the value of the aggregated point.
+                let Some(value) = (self.function)(sub_vec.clone()) else {
+                    return Err(TransformError::UnexpectedInput(anyhow!(
+                        "could not compute the aggregation for the sub_vec of {key:?}"
+                    )));
+                };
+
+                // Init the new point.
+                let new_point = MeasurementPoint::new_untyped(
+                    compute_min_timestamp(sub_vec[0].timestamp, self.interval),
+                    *metric_correspondence_table_read
+                        .get(&key.clone().0)
+                        .ok_or(TransformError::UnexpectedInput(anyhow!(
+                            "the metric ID {} is not known by the correspondence table",
+                            &key.clone().0.as_u64()
+                        )))?,
+                    key.clone().2,
+                    key.clone().1,
+                    value,
+                )
+                .with_attr_vec(
+                    sub_vec[0]
+                        .attributes()
+                        .map(|(key, value)| (key.to_owned(), value.clone()))
+                        .collect(),
+                );
+
+                // Push the new point to the result buffer.
+                aggregated_points.push(new_point);
+            }
+        }
+
+        measurements.merge(&mut aggregated_points);
+
+        Ok(())
+    }
+}
+
+impl Transform for AggregationTransform {
+    fn apply(&mut self, measurements: &mut MeasurementBuffer, _: &TransformContext) -> Result<(), TransformError> {
+        let metric_correspondence_table_clone = Arc::clone(&self.metric_correspondence_table.clone());
+        let metric_correspondence_table_read = (*metric_correspondence_table_clone)
+            .read()
+            .map_err(|_| anyhow!("could not read the metric correspondence table"))?;
```

Comment:
lock poisoning is a good reason to panic here.
Also, just the following is enough:
```suggestion
        let metric_correspondence_table = self.metric_correspondence_table.read()
          .expect("metric_correspondence_table lock poisoned");
```